File: vuln-sandbox/python/cwe-732/aws-s3-public-read-acl-safe.py
# SAFE: aws-s3-public-read-acl — S3 object uploaded without public-read ACL
# Rule: AwsS3PublicReadAcl | CWE-732 | Expected: TrueNegative

import boto3

s3 = boto3.client('s3')


def upload_private_file(bucket: str, key: str, data: bytes) -> None:
    """Upload a file to S3 with private access (no public ACL)."""
    # SAFE: no ACL parameter; bucket default (private) applies
    s3.put_object(
        Bucket=bucket,
        Key=key,
        Body=data,
        ServerSideEncryption='AES256',
    )


if __name__ == '__main__':
    upload_private_file('my-private-bucket', 'reports/2024-q1.pdf', b'report data')
    print('Uploaded with private access')

File: vuln-sandbox/node/cwe-732/aws-s3-public-read-acl-safe.js
// SAFE: aws-s3-public-read-acl — S3 object uploaded without public-read ACL
// Rule: AwsS3PublicReadAcl | CWE-732 | Expected: TrueNegative

const { S3Client, PutObjectCommand } = require('@aws-sdk/client-s3');

const client = new S3Client({ region: process.env.AWS_REGION });

async function uploadPrivateFile(bucket, key, body) {
  // SAFE: no ACL specified; bucket default (private) applies
  const command = new PutObjectCommand({
    Bucket: bucket,
    Key: key,
    Body: body,
    ServerSideEncryption: 'AES256',
  });
  return client.send(command);
}

uploadPrivateFile('my-private-bucket', 'reports/2024-q1.pdf', Buffer.from('report data'))
  .then(() => console.log('Uploaded privately'))
  .catch(console.error);
